cluster/examples/kubernetes/edgefs/csi/iscsi/edgefs-iscsi-csi-driver.yaml

```diff
@@ -0,0 +1,338 @@
+---
+
+kind: StatefulSet
+apiVersion: apps/v1beta1
+metadata:
+  name: edgefs-iscsi-csi-controller
+spec:
+  serviceName: edgefs-iscsi-csi-controller
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: edgefs-iscsi-csi-controller
+    spec:
+      serviceAccount: edgefs-iscsi-csi-controller-service-account
+      containers:
+        # csi-provisioner: sidecar container that watches Kubernetes PersistentVolumeClaim objects
+        # and triggers CreateVolume/DeleteVolume against a CSI endpoint
+        - name: csi-provisioner
+          image: quay.io/k8scsi/csi-provisioner:v1.0.1
+          imagePullPolicy: IfNotPresent
+          args:
+            - --connection-timeout=25s
+            - --provisioner=io.edgefs.csi.iscsi
+            - --csi-address=/var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy
+        # csi-attacher: sidecar container that watches Kubernetes VolumeAttachment objects
+        # and triggers ControllerPublish/Unpublish against a CSI endpoint
+        - name: csi-attacher
+          image: quay.io/k8scsi/csi-attacher:v1.0.1
+          imagePullPolicy: IfNotPresent
+          args:
+            - --v=3
+            - --csi-address=/var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy
+        - name: csi-snapshotter
+          image: quay.io/k8scsi/csi-snapshotter:v1.0.1
+          imagePullPolicy: IfNotPresent
+          args:
+            - --connection-timeout=25s
+            - --csi-address=/var/lib/csi/sockets/pluginproxy/csi.sock
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy
+        - name: driver
+          image: edgefs/edgefs-csi:latest
+          imagePullPolicy: Always
+          args:
+            - --role=controller
+            - --driverType=iscsi          
+            - --nodeid=$(KUBE_NODE_NAME)
+            - --endpoint=unix://csi/csi.sock
+            - --verbose=debug
+          env:
+            - name: KUBE_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /csi
+            - name: secret
+              mountPath: /config
+              readOnly: true
+      volumes:
+        - name: socket-dir
+          emptyDir:
+        - name: secret
+          secret:
+            secretName: edgefs-iscsi-csi-driver-config
+---
+
+
+# ---------------------------
+# Edgefs ISCSI CSI Node Server
+# ---------------------------
+#
+# Runs driver node server (driver + registrar) on each node
+#
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: edgefs-iscsi-csi-node-service-account
+  namespace: default # replace with non-default namespace name if needed
```

Comment:
I think it would be better to put everything here not in the default namespace by default.
As namespaces don't hit a clusters performance, having a separate one, e.g., edgefs-csi as the namespace name.